script/update_go_version.sh

```diff
@@ -11,6 +11,3 @@ cd $SDIR/..
 MAJOR_MINOR_VERSION=$(grep '^go' go.mod | cut -d' ' -f2 | cut -d. -f1-2)
 
 find ./ -type f -name "go.mod" -execdir go get go@$MAJOR_MINOR_VERSION \; -execdir go get toolchain@none \;
-
-GO_VERSION=$(grep '^go' go.mod | cut -d' ' -f2)
-echo $GO_VERSION > .go-version
```

Comment:
>.go-version is kept around because it's still used in one place: the version bump automation. I plan to raise an issue to just create a workflow that runs make update-go-version since we already handle everything in a make task.

Shouldn't aligning the `.go-version` with the version from `go.mod` still be kept until `.go-version` is fully removed?